src/util/bip32.rs

```diff
@@ -531,16 +531,14 @@ impl ExtendedPrivKey {
             network: network,
             depth: data[4],
             parent_fingerprint: Fingerprint::from(&data[5..9]),
-            child_number: child_number,
+            child_number:  endian::slice_to_u32_be(&data[9..13]).into(),
             chain_code: ChainCode::from(&data[13..45]),
             private_key: PrivateKey {
                 compressed: true,
                 network: network,
                 key: secp256k1::SecretKey::from_slice(
                     &data[46..78]
-                ).map_err(|e|
-                    base58::Error::Other(e.to_string())
-                )?,
+                ).expect("correct size"),
```

Comment:
Can the `from_slice()` only fail if size is incorrect? 
It looks like https://github.com/bitcoin-core/secp256k1/blob/9e5939d2848df5508e3a6eb038273b28d255248c/src/secp256k1.c#L550-L559 can fail if there is overflow or if secret key is zero.

Reply:
Fixed in the other branch and now fine after the rebase